src/chunkserver/network_worker_thread.cc

```diff
@@ -1256,96 +1307,100 @@ void worker_fwdwrite(csserventry *eptr) {
 			return;
 		}
 		stats_bytesout += i;
-		eptr->fwdstartptr += i;
-		eptr->fwdbytesleft -= i;
-	}
-	if (eptr->fwdbytesleft == 0) {
-		eptr->fwdinitpacket.clear();
-		eptr->fwdstartptr = NULL;
-		eptr->fwdmode = HEADER;
-		eptr->fwdinputpacket.bytesleft = 8;
-		eptr->fwdinputpacket.startptr = eptr->fwdhdrbuff;
-		eptr->fwdinputpacket.packet = NULL;
-		eptr->state = WRITEFWD;
+		eptr->fwdStartPtr += i;
+		eptr->fwdBytesLeft -= i;
+	}
+	if (eptr->fwdBytesLeft == 0) {
+		eptr->fwdInitPacket.clear();
+		eptr->fwdStartPtr = nullptr;
+		eptr->fwdMode = ChunkserverEntry::Mode::Header;
+		eptr->fwdInputPacket.bytesLeft = PacketHeader::kSize;
+		eptr->fwdInputPacket.startPtr = eptr->fwdHeaderBuffer;
+		eptr->fwdInputPacket.packet = nullptr;
+		eptr->state = ChunkserverEntry::State::WriteForward;
 	}
 }
 
-void worker_forward(csserventry *eptr) {
+void worker_forward(ChunkserverEntry *eptr) {
 	TRACETHIS();
 	int32_t i;
-	if (eptr->mode == HEADER) {
-		i = read(eptr->sock, eptr->inputpacket.startptr, eptr->inputpacket.bytesleft);
+	if (eptr->mode == ChunkserverEntry::Mode::Header) {
+		i = read(eptr->sock, eptr->inputPacket.startPtr,
```

Comment:
Same here.

Reply:
Applied here: https://github.com/leil-io/saunafs/commit/8f8dcf733ccd489fa6b9b4426b1ec5629500c839
Please notice this is out of the scope of this change.